vllm/sequence.py

```diff
@@ -867,3 +877,138 @@ def clone(
             num_lookahead_slots=self.num_lookahead_slots,
             running_queue_size=self.running_queue_size,
         )
+
+
+@dataclasses.dataclass(frozen=True)
+class ModelInput:
```

Comment:
Sounds good! I can create a base empty `ModelInput` class and then subclass with GPUModelInput, since that one has a bunch of shared fields with others.